Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs

```diff
@@ -772,60 +811,61 @@ private void ReconcilePointers()
             for (int i = 0; i < pointerRays.Length; i++)
             {
-                switch (pointerData.Pointer.SceneQueryType)
+                switch (pointer.SceneQueryType)
                 {
                     case SceneQueryType.SimpleRaycast:
                         if (MixedRealityRaycaster.RaycastSimplePhysicsStep(pointerRays[i], prioritizedLayerMasks, out physicsHit))
                         {
-                            UpdatePointerRayOnHit(pointerData, pointerRays, in physicsHit, i, rayStartDistance);
+                            UpdatePointerRayOnHit(pointerRays, in physicsHit, i, rayStartDistance, hit);
                             return;
                         }
                         break;
                     case SceneQueryType.BoxRaycast:
                         Debug.LogWarning("Box Raycasting Mode not supported for pointers.");
                         break;
                     case SceneQueryType.SphereCast:
-                        if (MixedRealityRaycaster.RaycastSpherePhysicsStep(pointerRays[i], pointerData.Pointer.SphereCastRadius, prioritizedLayerMasks, out physicsHit))
+                        if (MixedRealityRaycaster.RaycastSpherePhysicsStep(pointerRays[i], pointer.SphereCastRadius, prioritizedLayerMasks, out physicsHit))
                         {
-                            UpdatePointerRayOnHit(pointerData, pointerRays, in physicsHit, i, rayStartDistance);
+                            UpdatePointerRayOnHit(pointerRays, in physicsHit, i, rayStartDistance, hit);
                             return;
                         }
                         break;
                     case SceneQueryType.SphereOverlap:
-                        Collider[] colliders = UnityEngine.Physics.OverlapSphere(pointerData.Pointer.Rays[i].Origin, pointerData.Pointer.SphereCastRadius, ~UnityEngine.Physics.IgnoreRaycastLayer);
+                        Collider[] colliders = UnityEngine.Physics.OverlapSphere(pointer.Rays[i].Origin, pointer.SphereCastRadius, ~UnityEngine.Physics.IgnoreRaycastLayer);
 
                         if (colliders.Length > 0)
                         {
-                            Vector3 testPoint = pointerData.Pointer.Rays[i].Origin;
+                            Vector3 testPoint = pointer.Rays[i].Origin;
                             GameObject closest = null;
                             float closestDistance = Mathf.Infinity;
                             Vector3 objectHitPoint = testPoint;
 
                             foreach (Collider collider in colliders)
                             {
                                 // Policy: in order for an collider to be near interactable it must have
-                                // a NearIneractionGrabbable component on it
+                                // a NearInteractionGrabbable component on it.
+                                // FIXME: This is assuming only the grab pointer is using SceneQueryType.SphereOverlap. 
```

Comment:
> // FIXME: This is assuming only the grab pointer is using SceneQueryType.SphereOverlap.  [](start = 31, length = 89)

Can you add a little more context about this FIXME, or file an issue with tracking information on this? It's difficult to understand without additional context how I would approach fixing this, how I would verify it, etc.

Reply:
Extended the comment and created this ticket: https://github.com/Microsoft/MixedRealityToolkit-Unity/issues/3758